File: backend/springboot/src/main/java/tn/maiko26/springboot/models/ChatMessage.java
package tn.maiko26.springboot.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "chat_messages")
public class ChatMessage {
    @Id
    @Column(name = "message_id")
    private String id;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private ChatRoom room;

    @ManyToOne
    @JoinColumn(name = "sender_id", referencedColumnName = "email")
    private User sender;

    @Column(name = "content", nullable = false)
    private String content;

    @Column(name = "created_at", nullable = false, columnDefinition = "timestamp default now()")
    private Date createdAt;

    @Column(name = "read_by", columnDefinition = "text[] default '{}'::text[]")
    @ElementCollection
    private List<String> readBy = new ArrayList<>();
}

File: backend/dotnet/Models/TBD/ChatMessage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet.Models
{
    [Table("chat_messages")]
    public class ChatMessage
    {
        [Key]
        [Required]
        [Column("message_id")]
        public string Id { get; set; } = string.Empty;

        [Required]
        [ForeignKey("ChatRoom")]
        [Column("room_id")]
        public string RoomId { get; set; } = string.Empty;
        public ChatRoom ChatRoom { get; set; }

        [Required]
        [ForeignKey("User")]
        [Column("sender_id")]
        public string SenderId { get; set; } = string.Empty;
        public User Sender { get; set; }

        [Required]
        [Column("content")]
        public string Content { get; set; } = string.Empty;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("read_by")]
        public List<string> ReadBy { get; set; } = new List<string>();
    }
}